boost/vita.patch

```diff
@@ -0,0 +1,1473 @@
+-  && !defined(__CYGWIN__)
++  && !defined(__CYGWIN__) \
++  && !defined(__vita__)
+   // The type used for processing pipe readiness notifications.
+   class pipe_read_op;
+ 
+diff --git a/boost/asio/detail/socket_select_interrupter.hpp b/boost/asio/detail/socket_select_interrupter.hpp
+index 382fb3461..8d6606108 100644
+--- a/boost/asio/detail/socket_select_interrupter.hpp
++++ b/boost/asio/detail/socket_select_interrupter.hpp
+@@ -21,7 +21,8 @@
+ 
+ #if defined(BOOST_ASIO_WINDOWS) \
+   || defined(__CYGWIN__) \
+-  || defined(__SYMBIAN32__)
++  || defined(__SYMBIAN32__) \
++  || defined(__vita__)
+ 
+ #include <boost/asio/detail/socket_types.hpp>
+ 
+diff --git a/boost/asio/detail/socket_types.hpp b/boost/asio/detail/socket_types.hpp
+index 4099deec3..bd4fddd00 100644
+--- a/boost/asio/detail/socket_types.hpp
++++ b/boost/asio/detail/socket_types.hpp
+@@ -56,13 +56,18 @@
+ # endif // !defined(BOOST_ASIO_NO_DEFAULT_LINKED_LIBS)
+ # include <boost/asio/detail/old_win_sdk_compat.hpp>
+ #else
++#if !defined(__vita__)
+ # include <sys/ioctl.h>
+-# if (defined(__MACH__) && defined(__APPLE__)) \
++#endif
++# if defined(__vita__)
++#  include <boost/asio/detail/vita_compat.hpp>
++# endif
++# if ((defined(__MACH__) && defined(__APPLE__)) \
+    || defined(__FreeBSD__) || defined(__NetBSD__) \
+    || defined(__OpenBSD__) || defined(__linux__) \
+-   || defined(__EMSCRIPTEN__)
++   || defined(__EMSCRIPTEN__)) && !defined(__vita__)
+ #  include <poll.h>
+-# elif !defined(__SYMBIAN32__)
++# elif !defined(__SYMBIAN32__) && !defined(__vita__)
+ #  include <sys/poll.h>
+ # endif
+ # include <sys/types.h>
+@@ -75,15 +80,19 @@
+ #  include <sys/select.h>
+ # endif
+ # include <sys/socket.h>
++#if !defined(__vita__)
+ # include <sys/uio.h>
+ # include <sys/un.h>
++#endif
+ # include <netinet/in.h>
+ # if !defined(__SYMBIAN32__)
+ #  include <netinet/tcp.h>
+ # endif
+ # include <arpa/inet.h>
++#if !defined(__vita__)
+ # include <netdb.h>
+ # include <net/if.h>
++#endif
+ # include <limits.h>
+ # if defined(__sun)
+ #  include <sys/filio.h>
+@@ -291,11 +300,42 @@ typedef int socket_type;
+ const int invalid_socket = -1;
+ const int socket_error_retval = -1;
+ const int max_addr_v4_str_len = INET_ADDRSTRLEN;
+-#if defined(INET6_ADDRSTRLEN)
++#if defined(INET6_ADDRSTRLEN) && !defined(__vita__)
+ const int max_addr_v6_str_len = INET6_ADDRSTRLEN + 1 + IF_NAMESIZE;
+ #else // defined(INET6_ADDRSTRLEN)
+ const int max_addr_v6_str_len = 256;
+ #endif // defined(INET6_ADDRSTRLEN)
++#if defined(__vita__)
++typedef int ioctl_arg_type;
++typedef uint32_t u_long_type;
++typedef uint16_t u_short_type;
++struct in4_addr_type { u_long_type s_addr; };
++struct in4_mreq_type { in4_addr_type imr_multiaddr, imr_interface; };
++
++typedef in6_addr in6_addr_type;
++
++struct in6_mreq_type {
++    in6_addr_type ipv6mr_multiaddr;
++    unsigned long ipv6mr_interface;
++};
++
++typedef sockaddr socket_addr_type;
++typedef sockaddr_in sockaddr_in4_type;
```

Comment:
sometypes is defined in `netinet/in.h`

Reply:
I know. This comes from asio patch. I really don't want to rewrite it right now

Reply:
Also, which types?

Reply:
yeah... remake patch file and maintaining of it is really painful. because now we have to maintain both of boost and asio..

anyway i can't understand this part.
why you redefined type on asio repo instead of just including right header & reusing of it?

Reply:
for example, `sockaddr_in`

Reply:
it isn't redefined. it defines `sockaddr_in4_type` as `sockaddr_in` (which is defined in netinet/in.h)

Reply:
haha... i think, i forgot c language :'(